### PR TITLE
Added Kitchen tests and Travis

### PR DESCRIPTION
**Pull request by Martin819**

### CODE REVIEW

**Comment by cloudlab-autobuild** [comment]
Can one of the admins verify this patch?

File: docker/client/container.sls [hunk 1]
```diff
@@ -6,13 +6,13 @@ include:
 {%- for name, container in client.get('container', {}).iteritems() %}
   {%- set id = name %}
   {%- set required_containers = [] %}
-
+{%- if grains.get('virtual_subtype', None) not in ['Docker', 'LXC'] %}
```
Comment (fpytloun) [inline docker/client/container.sls]: Wrap with noservices instead

**Comment by fpytloun** [comment]
Test is failing

**Comment by epcim** [comment]
[**@fpytloun**](https://github.com/fpytloun) commented on [Apr 21, 2017, 7:53 PM GMT+2](https://github.com/salt-formulas/salt-formula-docker/pull/11#issuecomment-296259369 "2017-04-21T17:53:13Z - Replied by Github Reply Comments"):
> Test is failing

`- E: Unable to locate package python-docker` on trusty

File: docker/client/compose.sls [hunk 2]
```diff
@@ -97,12 +98,15 @@ docker_{{ app }}_pull:
     - watch:
       - file: docker_{{ app }}_env
       - file: docker_{{ app }}_compose
+    {%- if not grains.get('noservices','false')%}
```
Comment (epcim) [inline docker/client/compose.sls]: `('noservices','false')%}`
replace to
`('noservices', False) %}`

across modified files

**Review by epcim — APPROVED** [review]

File: metadata.yml [hunk 3]
```diff
@@ -1,3 +1,6 @@
 name: "docker"
 version: "0.2"
-source: "https://github.com/tcpcloud/salt-formula-docker"
+source: "https://github.com/salt-formulas/salt-formula-docker"
+dependencies:
+- name: linux
+  source: "https://github.com/salt-formulas/salt-formula-linux"
```
Comment (epcim) [inline metadata.yml:6]: @fpytloun here is something wrong, not in this PR but in general. The https://github.com/salt-formulas/salt-formula-docker/blob/be2b4a8938f1635284bd70f036b45e644836aa7b/metadata/service/host/vendor_repo_ubuntu_trusty.yml require docker repo, that's the only linux dependency. IMHO this yaml file should be removed from metadata.

Linux formula dependency might be worth-full in the future, to setup some "linux.system" features, etc..

**Review by epcim — APPROVED** [review]

**Pull request merged**